docs/driver-deployment/upgrade-guide.md

```diff
@@ -0,0 +1,138 @@
+# Upgrade Guide
+
+This guide provides instructions for upgrading the Scality S3 CSI Driver to newer versions.
+
+## Prerequisites
+
+Before upgrading, ensure all requirements outlined in the **[Prerequisites](prerequisites.md)** guide are met for the target version.
+
+## Pre-Upgrade Steps
+
+**Step 1. Set namespace variable:**
+
```

Comment:
Use `#.` rather than `Step X.` 
You can have sub steps numbered by indenting:
#. this is parent step 1
   #. indent this for sub-step 1
   #. indent for sub-step 2
#. this is parent step 2


Recommend not making steps bold. To much bold stuff removes it emphasis.

Reply:
We are using pure markdown here.
`#.` won't work in markdown, it will just make it <h1>...
<img width="419" alt="Screenshot 2025-06-25 at 21 34 04" src="https://github.com/user-attachments/assets/0ac5c74d-50f9-459c-8d4f-cadc62071681" />


Also if I do not make it bold this is what it looks like

<img width="1052" alt="Screenshot 2025-06-25 at 21 35 22" src="https://github.com/user-attachments/assets/0de7e6e7-9838-4f0a-82b4-38d1b49ecf0c" />


With bold:
<img width="1081" alt="Screenshot 2025-06-25 at 21 35 50" src="https://github.com/user-attachments/assets/a80ffa75-6611-4be1-b06b-e7dd81a96a28" />

I felt its better with bold to have a distinction from normal text. What do you think?